feat: add basic `head` command

dahlia opened the pull request:
This patch partially addresses issue #92.  It implements the very basic `head` command, with the following limitations:

- It can accept at most one file.
- The only options it supports are `-n`/`--lines`.
- The `-n`/`--lines` option cannot accept negative numbers.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/denoland/deno_task_shell?pullRequest=93) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/denoland/deno_task_shell?pullRequest=93) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/denoland/deno_task_shell?pullRequest=93) it.</sub>

dsherret commented on src/shell/commands/head.rs:
```diff
@@ -0,0 +1,239 @@
+        writer.write_all(b"\n")?;
+        lines += 1;
+      } else {
+        writer.write_all(&buf)?;
```
I think this needs to write only the size of the buffer and not the entire buffer which might contain data from a previous read at the end? We should extract out this function and add a unit test that triggers this scenario.

dsherret commented on src/shell/commands/head.rs:
```diff
@@ -0,0 +1,239 @@
+      let size = read(&mut buf)?;
+      if size == 0 {
+        break;
+      } else if let Some(line) = buf.split(|&b| b == b'\n').next() {
```
What happens to the rest of the lines in the buffer? I'm not sure how the tests work so I must be reading this wrong.

dsherret reviewed (changes requested):
Nice, thanks for this! I just have a few questions.

dahlia commented:
@dsherret  As you pointed out in your review, there were actually a bunch of bugs in copy_lines(), so I extracted it out to the module level and added unit tests for it to fix them.  Could you take another look at this?

dahlia commented:
Would there be anything left for me to fix?

dsherret commented:
@dahlia no, thanks! I just need to find the time to review the actual functionality. I'll get this in for Deno 1.38 which will be released in I think a month.

dahlia commented:
Thanks, I'm looking forward to it!

dsherret reviewed (approved):
LGTM. Thanks @dahlia!

dahlia commented:
Thank you!
